### PR TITLE
feat(dialog): Introduce a dedicated button to cancel a dialog

### PR DESCRIPTION
**Pull request by sebald**
# Description

Make it more convenient to have a button that closes the `<Dialog>`. With this, there is less need to use the child function to access the `close` method. Instead you can now use `<Button slot="close">` to render a close button.

# Reviewers:

@marigold-ui/developer
@marigold-ui/designer

### CODE REVIEW

File: docs/content/components/overlay/dialog/dialog-appearance.demo.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default (props: DialogProps) => (
         details pre-filled, except for the date.
       </Dialog.Content>
       <Dialog.Actions>
-        <Button variant="secondary">Cancel</Button>
+        <Dialog.Cancel>Cancel</Dialog.Cancel>
```
Comment (sarahgm) [inline docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: Cancel should than only be used for cancel actions? 

I think its strange to write a Dialog.Cancel and than also a normal Button inside the actions?

Comment (sarahgm) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: Also there must not be canceled written down

**Review by sarahgm — CHANGES_REQUESTED** [review]

File: packages/components/src/intl.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const intl = {
```
Comment (sarahgm) [inline packages/components/src/intl.ts:1]: do we need something like this? Should this decision be documented somewhere

**Review by sarahgm — CHANGES_REQUESTED** [review]

File: packages/components/src/Dialog/Dialog.test.tsx [hunk 3]
```diff
@@ -519,3 +519,54 @@ test('close state has a listener', async () => {
...
+    expect(cancel).toBeInTheDocument();
+  });
+
+  await user.click(cancel);
+  await waitFor(() => {
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+  });
+});
+
+test('render custom cancel button', async () => {
```
Comment (sarahgm) [inline packages/components/src/Dialog/Dialog.test.tsx:551]: Is it allowed to have a custom cancel button still? Like before? (anyways the test text must be adjusted)

**Review by sarahgm — DISMISSED** [review]

File: docs/content/components/overlay/dialog/dialog-appearance.demo.tsx (hunk 1 above)
Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: I added the `Cancel` to it does not accidentally get translated ;)

Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: > I think its strange to write a Dialog.Cancel and than also a normal Button inside the actions?

@sarahgm suggestions?

File: packages/components/src/intl.ts (hunk 2 above)
Comment (sebald) [reply packages/components/src/intl.ts:1]: Usually we use the translations from `react-aria`. Would a comment in the file be enough?

File: packages/components/src/Dialog/Dialog.test.tsx (hunk 3 above)
Comment (sebald) [reply packages/components/src/Dialog/Dialog.test.tsx:551]: What should be adjusted?

Comment (sebald) [reply packages/components/src/Dialog/Dialog.test.tsx:551]: What should be adjusted?

Comment (sarahgm) [reply packages/components/src/Dialog/Dialog.test.tsx:551]: the test text ('render custom cancel button) because its not custom

File: docs/content/components/overlay/dialog/dialog-appearance.demo.tsx (hunk 1 above)
Comment (sarahgm) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: I'm not sure if I would call it Dialog.Cancel, I think its also ok to have a dedicated cancel button for everything

File: packages/components/src/Dialog/Dialog.test.tsx (hunk 3 above)
Comment (sebald) [reply packages/components/src/Dialog/Dialog.test.tsx:551]: It's a custom label. Very nitpicky 😜

File: docs/content/components/overlay/dialog/dialog-appearance.demo.tsx (hunk 1 above)
Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: But for everything it not possible, this is a "dedicated" button, because you no longer have to use the callback function so you can access the `close` function. Can also name it close instead? But I wasn't sure if this is confusing with the `closeButton` prop on the `<Dialog>`

Comment (sarahgm) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: Just my opinion :P I mean this:
```tsx
<Dialog.Actions>
	<Button variant="close"/>
	<Button variant="primary"/>
</Dialog.Actions>
```
But its ok

Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: Yeah, but having the `Button` depend on the `OverlayContext` seems really bad.

Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: Any thoughts @aromko @OsamaAbdellateef

Comment (aromko) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: For me it's fine. Also the fallback with button.cancel.

Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-appearance.demo.tsx:14]: So merge it? :P

**Review by aromko — DISMISSED** [review]

**Comment by sebald** [comment]
> @sebald Merging is blocked 😬

Hu. Why? Review is missing. Right?

**Comment by aromko** [comment]
> > @sebald Merging is blocked 😬
> 
> Hu. Why? Review is missing. Right?

No. I think cause of requested changes of @sarahgm .

**Comment by sebald** [comment]
@aromko yes, that is why I asked in the thread. you need to resolve it too :P

**Review by aromko — APPROVED** [review]

File: docs/content/components/overlay/dialog/dialog-from-menu.demo.tsx [hunk 4]
```diff
@@ -32,10 +32,10 @@ export default () => {
                 be undone.
               </Dialog.Content>
               <Dialog.Actions>
-                <Button size="small" variant="secondary" onPress={close}>
+                <Button variant="secondary" slot="close">
                   Cancel
                 </Button>
-                <Button size="small" variant="primary" onPress={close}>
+                <Button variant="primary" onPress={close}>
```
Comment (aromko) [inline docs/content/components/overlay/dialog/dialog-from-menu.demo.tsx:38]: The `onPress={close}` isn't here necessary or? So maybe as example and alert like `onPress={() => alert('deleted!'}`

**Review by aromko — DISMISSED** [review]

File: .changeset/few-points-vanish.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+"@marigold/docs": minor
+"@marigold/components": minor
+---
+
+feat(dialog): Introduce a dedicated button to close a dialog
+
+Make it more convenient to have a button that closes the `<Dialog>`. With this, there is less need to use the child function to access the `close` method. Instead you can now use `<Button slote="close">` to render a close button.
```
Comment (sarahgm) [inline .changeset/few-points-vanish.md:8]: slot not slote :)

File: packages/components/src/Dialog/DialogActions.tsx [hunk 6]
```diff
@@ -12,9 +23,26 @@ export interface DialogActions {
...
   return (
     <div className={cn('[grid-area:actions]', classNames.actions)}>
-      {children}
+      <Provider
+        values={[
+          [
+            ButtonContext,
+            {
+              slots: {
+                [DEFAULT_SLOT]: {},
```
Comment (sarahgm) [inline packages/components/src/Dialog/DialogActions.tsx:37]: what is a default_slot?

File: packages/components/src/Select/intl.ts [hunk 7]
```diff
@@ -1,8 +0,0 @@
-export const messages = {
```
Comment (sarahgm) [inline packages/components/src/Select/intl.ts:1]: why is this deleted?

**Review by sarahgm — CHANGES_REQUESTED** [review]

File: packages/components/src/Dialog/DialogActions.tsx (hunk 6 above)
Comment (sebald) [reply packages/components/src/Dialog/DialogActions.tsx:37]: Needed so you can but a button on there without a slot :) 

https://react-spectrum.adobe.com/react-aria/advanced.html#default-slot

File: packages/components/src/Select/intl.ts (hunk 7 above)
Comment (sebald) [reply packages/components/src/Select/intl.ts:1]: we did not use this for a long time (was used when we used hooks)

File: docs/content/components/overlay/dialog/dialog-from-menu.demo.tsx (hunk 4 above)
Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-from-menu.demo.tsx:38]: but the modal will not close then 🤔

Comment (aromko) [reply docs/content/components/overlay/dialog/dialog-from-menu.demo.tsx:38]: ok true. In Storybook and also in the documentation is no example of doing an action with the primary button and then close the dialog 🤔 At least we should add one or?

**Comment by sebald** [comment]
Hm this is approved. Merge? @sarahgm @aromko

File: docs/content/components/overlay/dialog/dialog-from-menu.demo.tsx (hunk 4 above)
Comment (sebald) [reply docs/content/components/overlay/dialog/dialog-from-menu.demo.tsx:38]: I thought this is clear 😄
